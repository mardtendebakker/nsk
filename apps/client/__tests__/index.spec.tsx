import React from 'react';
import { render } from '@testing-library/react';
import Dashboard from '../pages/index';

const mockRouter = {
  push: jest.fn(),
};

jest.mock('next/router', () => ({
  useRouter: () => mockRouter,
}));

jest.mock('../layouts/dashboard', () => ({ children }:{ children:JSX.Element }) => children);
jest.mock('../components/dashboard/analytics', () => ({ children }:{ children:JSX.Element }) => children);

const mockUseSecurity = {
  signOut: jest.fn(() => Promise.resolve()),
  state: { get user() { return null; } },
};

jest.mock('../hooks/useSecurity', () => jest.fn(() => mockUseSecurity));

describe('Dashboard component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Dashboard />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('shows username', () => {
    jest.spyOn(mockUseSecurity.state, 'user', 'get').mockReturnValue({ username: 'username' });
    const { getByText } = render(<Dashboard />);
    expect(getByText(/username/i)).toBeDefined();
  });

  it('shows username', () => {
    const { queryByText } = render(<Dashboard />);
    expect(queryByText('username')).toBeNull();
  });
});
