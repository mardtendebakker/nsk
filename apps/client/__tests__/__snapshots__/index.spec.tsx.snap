// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component matches snapshot 1`] = `
<DocumentFragment>
  <h3
    class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
  >
    hello, !
  </h3>
  <div
    class="MuiBox-root css-6s73tp"
  />
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-1ufd20l-MuiCardContent-root"
    >
      <div
        class="MuiBox-root css-1hac7jn"
      >
        <p
          class="MuiTypography-root MuiTypography-inherit css-adfmws-MuiTypography-root"
        >
          Test Test 1
        </p>
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          € 100,00
        </h3>
      </div>
      <div
        class="MuiBox-root css-1hac7jn"
      >
        <p
          class="MuiTypography-root MuiTypography-inherit css-adfmws-MuiTypography-root"
        >
          Test Test 2
        </p>
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          € 100,00
        </h3>
      </div>
      <div
        class="MuiBox-root css-1hac7jn"
      >
        <p
          class="MuiTypography-root MuiTypography-inherit css-adfmws-MuiTypography-root"
        >
          Test Test 3
        </p>
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          € 100,00
        </h3>
      </div>
      <div
        class="MuiBox-root css-1hac7jn"
      >
        <p
          class="MuiTypography-root MuiTypography-inherit css-adfmws-MuiTypography-root"
        >
          Test Test 4
        </p>
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          € 100,00
        </h3>
      </div>
      <div
        class="MuiBox-root css-1hac7jn"
      >
        <p
          class="MuiTypography-root MuiTypography-inherit css-adfmws-MuiTypography-root"
        >
          Test Test 5
        </p>
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          € 100,00
        </h3>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-19ysx2c"
  >
    <div
      class="MuiBox-root css-odlrzr"
    >
      <div
        class="MuiBox-root css-6s73tp"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-fjo7g7-MuiTypography-root"
          >
            locationCapacity
          </h5>
          <table
            class="MuiTable-root css-rqglhn-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
                  scope="col"
                >
                  locationName
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
                  scope="col"
                >
                  status
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
                  scope="col"
                >
                  quantity
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
                  scope="col"
                >
                  lastOrder
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            />
          </table>
        </div>
      </div>
      <div
        class="MuiBox-root css-6s73tp"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-fjo7g7-MuiTypography-root"
          >
            ResourceManagement
          </h5>
          <table
            class="MuiTable-root css-rqglhn-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
                  scope="col"
                >
                  name
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
                  scope="col"
                >
                  email
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
                  scope="col"
                >
                  numberOfTasks
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
                  scope="col"
                >
                  status
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
                  scope="col"
                >
                  role
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            />
          </table>
        </div>
      </div>
      <div
        class="MuiBox-root css-6s73tp"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-fjo7g7-MuiTypography-root"
          >
            dueToday
          </h5>
          <table
            class="MuiTable-root css-rqglhn-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
                  scope="col"
                >
                  taskName
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
                  scope="col"
                >
                  productName/type
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
                  scope="col"
                >
                  dueBy
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            />
          </table>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-1wzezhk"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          >
            upcomingDeliveries 1
          </h5>
          <div
            class="MuiBox-root css-10awhm1"
          />
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiBox-root css-gg4vpm"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
              >
                test test 12345 1
              </h5>
              <div
                class="MuiBox-root css-1qadj9d"
              >
                inTransit
              </div>
            </div>
            <span
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="0"
              class="MuiLinearProgress-root MuiLinearProgress-colorInherit MuiLinearProgress-determinate css-18za701-MuiLinearProgress-root"
              role="progressbar"
            >
              <span
                class="MuiLinearProgress-bar MuiLinearProgress-barColorInherit MuiLinearProgress-bar1Determinate css-1wqgwi1-MuiLinearProgress-bar1"
                style="transform: translateX(-100%);"
              />
            </span>
            <div
              class="MuiBox-root css-i51og3"
            >
              <ul
                style="list-style-type: none;"
              >
                <li>
                  <div
                    class="MuiBox-root css-70qvj9"
                  >
                    <div
                      class="MuiBox-root css-lhrrb6"
                    />
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    >
                      Test test
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-1twekll-MuiTypography-root"
                    >
                      0:00 pm
                    </p>
                  </div>
                  <div
                    class="MuiBox-root css-h0ro49"
                  />
                </li>
                <li>
                  <div
                    class="MuiBox-root css-70qvj9"
                  >
                    <div
                      class="MuiBox-root css-lhrrb6"
                    />
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    >
                      Test test
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-1twekll-MuiTypography-root"
                    >
                      0:00 pm
                    </p>
                  </div>
                  <div
                    class="MuiBox-root css-h0ro49"
                  />
                </li>
                <li>
                  <div
                    class="MuiBox-root css-70qvj9"
                  >
                    <div
                      class="MuiBox-root css-lhrrb6"
                    />
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    >
                      Test test
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-1twekll-MuiTypography-root"
                    >
                      0:00 pm
                    </p>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
